diff -up samples/opencl/cl/app/TemplateC.orig//Makefile samples/opencl/cl/app/TemplateC//Makefile
--- samples/opencl/cl/app/TemplateC.orig//Makefile	2011-07-18 21:09:23.000000000 +0300
+++ samples/opencl/cl/app/TemplateC//Makefile	2011-10-05 23:48:36.345172884 +0300
@@ -22,5 +22,7 @@ EXE_TARGET_INSTALL   	= TemplateC
 FILES 	= TemplateC
 CLFILES	= TemplateC_Kernels.cl
 
+LLIBS  += SDKUtil
+
 include $(DEPTH)/make/openclsdkrules.mk 
 
diff -up samples/opencl/cl/app/TemplateC.orig//TemplateC.cpp samples/opencl/cl/app/TemplateC//TemplateC.cpp
--- samples/opencl/cl/app/TemplateC.orig//TemplateC.cpp	2011-07-18 21:09:23.000000000 +0300
+++ samples/opencl/cl/app/TemplateC//TemplateC.cpp	2011-10-05 23:50:57.353178673 +0300
@@ -339,11 +339,8 @@ initializeCL(void)
     // Load CL file, build CL program object, create CL kernel object
     /////////////////////////////////////////////////////////////////
     const char * filename  = "TemplateC_Kernels.cl";
-    std::string  sourceStr;
-    status = convertToString(filename, sourceStr);
-    if(status != CL_SUCCESS)
-        return 1;
-
+    std::string  sourceStr = streamsdk::SDKCommon::getPath();
+    sourceStr.append(filename);
     const char * source    = sourceStr.c_str();
     size_t sourceSize[]    = { strlen(source) };
 
diff -up samples/opencl/cl/app/TemplateC.orig//TemplateC.hpp samples/opencl/cl/app/TemplateC//TemplateC.hpp
--- samples/opencl/cl/app/TemplateC.orig//TemplateC.hpp	2011-07-18 21:09:23.000000000 +0300
+++ samples/opencl/cl/app/TemplateC//TemplateC.hpp	2011-10-05 23:49:27.433174980 +0300
@@ -102,6 +102,7 @@ jurisdiction and venue of these courts.
 #include <iostream>
 #include <string>
 #include <fstream>
+#include <SDKCommon.hpp>
 
 
 /*** GLOBALS ***/
